### ISSUE TITLE
Collection/Compact on ActionController::Parameters 6.0

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#11066]
I'm seeing this false positive. 
```
lib/blacklight/parameters.rb:14:14: C: [Correctable] Style/CollectionCompact: Use compact instead of reject { |_k, v| v.nil? }.
      params.reject { |_k, v| v.nil? }
             ^^^^^^^^^^^^^^^^^^^^^^^^^
```
When I make this change it fails because ActionController::Parameters 6.0 does not have a compact method.  It doesn't seem like one can infer that a `compact` method exists on an object just because a `.reject` method exists.

**Comment by koic** [comment]
`Style/CollectionCompact` cop is already marked as unsafe. What could be improved is to update the documentation in the `@safety` section.

**Comment by PolyCole** [comment]
Hey there. If no one has expressed interest, I'd love to take a look at this issue. From what I can gather, I'll be updating [this part of the docs](https://github.com/rubocop/rubocop/blob/38e8511f78c86fa1c3109311f2c872b9e69c3b10/lib/rubocop/cop/style/collection_compact.rb#L9) to note that a `compact` method might not exist despite Rubocop highlighting the line?
